src/stores/settingsStore.ts

```
@@ -30,6 +36,43 @@ export const useSettingsStore = create<SettingsStore>()(
+      clearSelectedActions: () =>
+        set((state) => ({
+          settings: { ...state.settings, selectedActions: {} },
+        })),
+      getSelectedActionsByType: (type: string): Record<string, ActionEntry> => {
+        const { selectedActions = {} } = useSettingsStore.getState().settings;
+        const actions: Record<string, ActionEntry> = selectedActions || {};
+        return Object.entries(actions)
+          .filter(([, entry]: [string, ActionEntry]) => entry.type === type)
+          .reduce(
+            (acc: Record<string, ActionEntry>, [key, entry]: [string, ActionEntry]) => ({
+              ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)